daemon/ucrednet_test.go

```diff
@@ -0,0 +1,109 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"errors"
+	"net"
+	"path/filepath"
+	sys "syscall"
+
+	"gopkg.in/check.v1"
+)
+
+type ucrednetSuite struct {
+	ucred *sys.Ucred
+	err   error
+}
+
+var _ = check.Suite(&ucrednetSuite{})
+
+func (s *ucrednetSuite) getUcred(fd, level, opt int) (*sys.Ucred, error) {
+	return s.ucred, s.err
+}
+
+func (s *ucrednetSuite) SetUpSuite(c *check.C) {
+	getUcred = s.getUcred
+}
+
+func (s *ucrednetSuite) TearDownTest(c *check.C) {
+	s.ucred = nil
+	s.err = nil
+}
+func (s *ucrednetSuite) TearDownSuite(c *check.C) {
+	getUcred = sys.GetsockoptUcred
+}
+
+func (s *ucrednetSuite) TestAcceptConnRemoteAddrString(c *check.C) {
```

Comment:
need a test about the non-unix socket case,  also as I said I think the feature/branch would be more complete/consistent if a helper to extract uids from addresses was added